### PR TITLE
Fix race in containerd metrics that can lead to missing tags

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Fix race in containerd metrics that can lead to missing tags

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Test is very difficult as issue only appears in specific conditions, though we're somewhat able to reproduce on internal infrastructure.

### CODE REVIEW

**Review by xornivore — COMMENTED** [review]
Any way we could simulate the race in a test?

File: releasenotes/notes/containerd-missing-tags-7128154ecb803b3b.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Fixes submit of containers blkio metrics that may modify array after being already used by aggregator. Can cause missing tags on containerd.* metrics
```
Comment (olivielpeau) [inline releasenotes/notes/containerd-missing-tags-7128154ecb803b3b.yaml:4]:
```suggestion
    Fixes submission of containers blkio metrics that may modify array after being already used by aggregator. Can cause missing tags on containerd.* metrics
```

**Comment by vboulineau** [comment]
> Any way we could simulate the race in a test?

Nothing comes to my mind right now. Technically we should check every metric...

**Review by olivielpeau — APPROVED** [review]
LGTM, just a nit on the release note

**Review by xornivore — APPROVED** [review]

**Pull request merged**